src/main.etk

```diff
@@ -7,9 +7,13 @@
         98304
 %end
 
+%macro function_sig() 
+        push4 0x9507d39a
+%end
+
 %macro get_input()
-        %push0()
-        calldataload
+        push1 4      # [4]
```

Comment:
oh this is 4 words, you'll actually need to load 1 word and mask out the upper 4 bytes. you can see how to do that here: https://github.com/lightclient/erc20/blob/7b0d75d60f4e746c697a9ff88c62da411ac3f3b5/src/jump_table.etk#L20-L23

Reply:
Calldataload puts 32 bytes from calldata on stack, and item on stack is the offset?

Reply:
> oh this is 4 words

Nope, the calldataload offset is a byte-offset, not word-offset
```golang
func opCallDataLoad(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
	x := scope.Stack.peek()
	if offset, overflow := x.Uint64WithOverflow(); !overflow {
		data := getData(scope.Contract.Input, offset, 32)
		x.SetBytes(data)
```
So it's correct

Reply:
ah you're right i got confused with how i was mloading the data in the other project!